src/main/scala/com/fulcrumgenomics/fasta/UpdateFastaContigNames.scala

```diff
@@ -24,70 +24,157 @@
  @arg(flag='d', doc="The path to the sequence dictionary with contig aliases.") val dict: PathToSequenceDictionary,
  @arg(flag='o', doc="Output FASTA.") val output: PathToFasta,
  @arg(flag='l', doc="Line length or sequence lines.") val lineLength: Int = 100,
- @arg(doc="Skip missing source contigs.") val skipMissing: Boolean = false
+ @arg(doc="Skip missing source contigs.") val skipMissing: Boolean = false,
+ @arg(doc="Sort the contigs based on the input sequence dictionary") sortByDict: Boolean = false,
+ @arg(doc="Add sequences from this FASTA when contigs in the sequence dictionary are missing from the input FASTA")
+ defaultContigs: Option[PathToFasta] = None
 ) extends FgBioTool with LazyLogging {
+  import com.fulcrumgenomics.fasta.Converters.FromSAMSequenceDictionary
 
-  Io.assertReadable(input)
   Io.assertReadable(Seq(input, dict))
+  defaultContigs.foreach(Io.assertReadable)
   Io.assertCanWriteFile(output)
 
+  /** Little class to store the sequence metadata and reference sequence to output.
+    *
+    * The reference sequence to output is stored as a method, so we can lazily retrieve it and not incur large memory
+    * overhead.
+    *
+    * @param info the sequence metadata to output
+    * @param sequence the reference sequence to output
+    */
+  private class OutputSequence(val info: SequenceMetadata, sequence: => ReferenceSequence) {
+    /** Writes the output sequence */
+    def write(writer: BufferedWriter, progress: ProgressLogger): Unit = {
+      val ref = sequence
+      writer.append('>').append(info.name).append('\n')
+      val bases = ref.getBases
+      var baseCounter = 0
+      forloop(from = 0, until = bases.length) { baseIdx =>
+        writer.write(bases(baseIdx))
+        progress.record(info.name, baseIdx + 1)
+        baseCounter += 1
+        if (baseCounter >= lineLength) {
+          writer.newLine()
+          baseCounter = 0
+        }
+      }
+      if (baseCounter > 0) writer.newLine()
+    }
+  }
+
   override def execute(): Unit = {
     val progress = ProgressLogger(logger, noun="bases", verb="written", unit=10e7.toInt)
     val dict     = SequenceDictionary(this.dict)
-    val refFile  = ReferenceSequenceFileFactory.getReferenceSequenceFile(input, true, true)
     val out      = Io.toWriter(output)
 
-    val srcContigs = refFile.getSequenceDictionary match {
-      case null =>
-        require(refFile.isIndexed,
-          "Reference sequence file must have a sequence dictionary or be indexed.  Try 'picard CreateSequenceDictionary' or 'samtools faidx <ref.fasta>'.")
-        new FastaSequenceIndex(ReferenceSequenceFileFactory.getFastaIndexFileName(this.input)) map(_.getContig)
-      case dict => dict.getSequences.map(_.getSequenceName)
+    val (srcDict, srcRefFile) = getDictAndFile(fasta=this.input)
+    val defaultDictAndRefFile = this.defaultContigs.map(fasta => getDictAndFile(fasta=fasta))
+
+    // Make sure the default FASTA and the input dict are the same
```

Comment:
Why make this a requirement?  E.g. what if I have a nicely ordered dict that I made, but my default fasta is hastily sourced from UCSC and is in lexicographic order?  Maybe just warn for this?  Or fail later if there's a sequence needed that can't be found within the default?

Reply:
The comment may be misleading.  I mean to say:
> Make sure the default FASTA and the **_default_** input dict are the same, as the latter should be built from the former